libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -279,7 +282,9 @@ bool VoteManager::addVerifiedVote(const std::shared_ptr<Vote>& vote) {
           {two_plus_one_voted_block_type, std::make_pair(vote->getBlockHash(), vote->getStep())});
 
       // Save only current pbft period & round 2t+1 votes bundles into db
-      if (vote->getPeriod() == current_pbft_period_ && vote->getRound() == current_pbft_round_) {
+      // Cert votes are saved once the pbft block is pushed in the chain
```

Comment:
1 potential problem I see here is that if we are not able to push the block due to missing tx/dag block and we want to restart the node to reset caches etc... we will lose those votes

Reply:
if we came to the point of having 2t+1 cert votes without having a pbft block or dag block there is something else seriously wrong on this specific node. Since every node saves its own votes anyhow I considered this scenario and from the two options:
1. As implemented in PR accepting the loss of votes in the scenario
2. Saving 2t+1 votes before pushing a new pbft block to the chain and then saving it in another storage after pushing block to the chain, decrease performance since we need two db saves and a copy

I would choose 1. but if everyone think 2. is a safer option I am not strongly set against it. But if we go to option 2. one could argue option 3. where we save all cert votes all the time not to lose any even before we have 2t+1. Because currently we only save to db when we reach 2t+1. If we restart before reaching 2t+1 we do not save and we lose these votes